src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -808,35 +808,46 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			return true;
 		}
 		let azureResource = this.getAzureResourceForConnection(connection);
-		let accounts = (await this._accountManagementService.getAccounts()).filter(a => a.key.providerId.startsWith('azure'));
-		if (accounts && accounts.length > 0) {
+		const accounts = await this._accountManagementService.getAccounts();
+		const azureAccounts = accounts.filter(a => a.key.providerId.startsWith('azure'));
+		if (azureAccounts && azureAccounts.length > 0) {
 			let accountName = (connection.authenticationType === Constants.azureMFA) ? connection.azureAccount : connection.userName;
-			let account = find(accounts, account => account.key.accountId === accountName);
+			let account = find(azureAccounts, account => account.key.accountId === accountName);
 			if (account) {
+				this._logService.debug(`Getting security token for Azure account ${account.key.accountId}`);
 				if (account.isStale) {
+					this._logService.debug(`Account is stale - refreshing`);
 					try {
 						account = await this._accountManagementService.refreshAccount(account);
-					} catch {
+					} catch (err) {
+						this._logService.info(`Exception refreshing stale account : ${err}`);
```

Comment:
You may want to JSON.stringify this

Reply:
It'll print it out in the format `Error : <Error Text here>` already - do you think this would contain additional JSON content that needs to be printed specially?

Reply:
It's just so in case somehow a JSON object does end up in the error message, we can still get useful information. JSON.stringify also handles strings properly so it's just something that would make it work for both cases.

Reply:
Guess it can't hurt. We're so inconsistent with how we print debug strings though - let's try to think of ways to actually enforce this stuff or make some helper to do that for us so it's easy to apply across the board. Maybe just a simple helper function that takes in a parameterized string and then applies JSON.stringify to all the params?

```
function stringifyMessage(msg: string, ...args: any[]) { 
   return strings.format(msg, args.map(a => JSON.stringify(a)));
}
```

Reply:
Probably a good idea! I really dislike how JS just handles printing JSON objects by default.

Reply:
Bleh - not really a fan of how this reads :

![image](https://user-images.githubusercontent.com/28519865/81595711-cd1f6700-9377-11ea-8ba4-e8820f021f03.png)

What about just using the [toErrorMessage ](https://github.com/microsoft/azuredatastudio/blob/master/src/vs/base/common/errorMessage.ts#L42)that VS Code already has? That'll print errors nicely : 

![image](https://user-images.githubusercontent.com/28519865/81596188-7bc3a780-9378-11ea-804a-f0be18591640.png)

I like this better since currently our ES Lint rules enforce only throwing an Error anyways.